### PR TITLE
feat: Workspace v3

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Motivation

Given my experience with onboarding new members of the team I continously find me in a situation where I have to explain the logic of the code to my peers. This process seems to repeat more than necessary due to what I would describe as unnecessary complexity. The motivation behind this PR has been to reduce complexity and increase maintainability. By simplifying code and increasing test coverage. So far I have had to personally vet every single PR carefully, the long-term goal would be to have such a clean and clear codebase that changes could be made and reviewed by another peer without me being involved.

`Any fool can write code that a computer can understand. Good programmers write code that humans can understand.` – Martin Fowler

# Changes

- [x] `workspace core` deprecated
- [x] `observable proxy` deprecated
- [x] WorkspaceContext introduced
- [x] useWorkspace hook
- [x] Fixed selection load from url
- [x] Implemented selection/deselection for ag-grid
- [x] Implemented selection/deselection for garden
- [x] add suspense to sidesheet
- [x] add integration tests for Workspace
- [x] API for selection is deprecated, call node.setSelected(value) instead
- [x] No refetching on filters changed
- [x] Initial grouping undefined for all requests

# Long term goals
- [ ] Internalize Workspace-react
- [ ] Rename Workspace-fusion to Workspace
- [ ] Deprecate workspace-react package
- [ ] Remove [rxjs](https://rxjs.dev)

# Fixes
Closes: #433 
Closes: #434 
Closes: #435 
Closes: #436
Closes: #437
Closes: #438

# Testing 

The mission of integration testing in this lib is to ensure that the external facade of Workspace is intact between changes.
This means that the following functionality should always work when doing changes.
I think keeping all tests in a single package makes sense for several reasons:
- You are testing the external facade of Workspace
- Checking the test coverage and adding new tests is easy and centralized.

Test coverage:
- [x] Opening sidesheet if `item` queryparam is present.
- [x] Opening the correct tab if `tab` queryparam is present.
- [x] Api calls for garden are called when the garden tab is mounted.
- [x] KPI api call
- [x] Filter api call
- [x] Search filter retriggers fetch

I think the most important tests have been covered for now. I dont want to bury this project in unncessary tests. If a feature keeps breaking over time with changes a new test will be added

### CODE REVIEW

**Comment by Gustav-Eikaas** [comment]
Alpha 2.0 verified good with Workorder app in CC

Fixes
- [ ] Triple backspace bug due to history overwrite
- [ ] Sidesheet closing after initial load
- [ ] Deselection in grid and garden

**Comment by Gustav-Eikaas** [comment]
Possibly move isolated tests to their respective packages. Like garden item rendering etc... and keep integration tests in a separate package

**Comment by Gustav-Eikaas** [comment]
Should write a few more tests and this should be ready for merging

**Comment by Gustav-Eikaas** [comment]
Seems like react-testing-library doesnt do layout so testing sizes wont be possible. Something like cyprus would be necessary. I would have wanted to write tests ensuring virtualization works as this is prone to breaking with the smallest changes.

File: packages/garden/src/lib/components/GardenItemContainer/GardenItemContainer.tsx [hunk 1]
```diff
@@ -293,22 +286,21 @@ export const GardenItemContainer = <TData extends Record<PropertyKey, unknown>,
...
                 data={item}
-                isSelected={selectedIds.includes(getIdentifier(item))}
+                isSelected={selection === getIdentifier(item)}
                 onClick={() => {
-                  controller.selectedNodes.setValue([getIdentifier(item)]);
-                  controller.clickEvents.onClickItem && controller.clickEvents.onClickItem(item);
+                  onClickItem(item);
                 }}
                 width={itemWidth}
                 depth={0}
```
Comment (Gustav-Eikaas) [inline packages/garden/src/lib/components/GardenItemContainer/GardenItemContainer.tsx:306]: Need to verify if this is still needed without recursive grouping. I dont think it is

Comment (Gustav-Eikaas) [reply packages/garden/src/lib/components/GardenItemContainer/GardenItemContainer.tsx:306]: depth={0}

**Comment by Gustav-Eikaas** [comment]
> Possibly move isolated tests to their respective packages. Like garden item rendering etc... and keep integration tests in a separate package

Decided against it because the objective of tests here is to check the external facade of workspace not the components isolated. Also the lines could become blurry between where to locate the tests. Possibly leading to confusion and duplicate tests

File: packages/workspace-fusion-integration-test/tests/Filter.test.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { Workspace } from '@equinor/workspace-fusion';
+import { act, render } from '@testing-library/react';
+import { workspaceOptions } from '../config/workspaceOptions';
+import { fakeTimeout } from '../utils/fakeTimeout';
+import { getMockGardenConfig } from '../config/mockGardenConfig';
+import { getMockFilterConfig } from '../config/mockFilterConfig';
+
+describe('Testing garden integration', () => {
```
Comment (Gustav-Eikaas) [inline packages/workspace-fusion-integration-test/tests/Filter.test.tsx:9]: typo filter

**Review by MagnusGjosund — APPROVED** [review]

**Comment by MagnusGjosund** [comment]
@Gustav-Eikaas test mention

**Pull request merged**